src/main/java/org/jenkinsci/plugins/DependencyTrack/DependencyTrackPublisher.java

```diff
@@ -178,91 +179,90 @@ public void perform(@Nonnull final Run<?, ?> build,
-                return;
-            }
-            final ApiClient.UploadResult uploadResult = apiClient.upload(projectId, projectName, projectVersion,
-                    artifactFilePath, isScanResult, autoCreateProject);
 
-            if (!uploadResult.isSuccess()) {
-                build.setResult(Result.FAILURE); //todo: make configurable
-                return;
-            }
+        if (!filePath.exists()) {
+            logger.log(Messages.Builder_Artifact_NonExist());
+            throw new AbortException("Nonexistent artifact");
+        }
+        final ApiClient.UploadResult uploadResult = apiClient.upload(projectId, projectName, projectVersion,
+                artifactFilePath, isScanResult, autoCreateProject);
 
-            if (synchronous && isScanResult) {
-                logger.log(Messages.Builder_Artifact_NonBomSync());
-                return;
-            }
+        if (!uploadResult.isSuccess()) {
+            throw new AbortException("Dependency Track server upload failed");
+        }
+
+        if (synchronous && isScanResult) {
+            logger.log(Messages.Builder_Artifact_NonBomSync());
+            return;
+        }
 
-            if (uploadResult.getToken() != null && synchronous) {
-                final long timeout = System.currentTimeMillis() + (60000L * getDescriptor().getDependencyTrackPollingTimeout());
+        if (uploadResult.getToken() != null && synchronous) {
+            final long timeout = System.currentTimeMillis() + (60000L * getDescriptor().getDependencyTrackPollingTimeout());
+            Thread.sleep(10000);
+            logger.log(Messages.Builder_Polling());
+            while (apiClient.isTokenBeingProcessed(uploadResult.getToken())) {
                 Thread.sleep(10000);
-                logger.log(Messages.Builder_Polling());
-                while (apiClient.isTokenBeingProcessed(uploadResult.getToken())) {
-                    Thread.sleep(10000);
-                    if (timeout < System.currentTimeMillis()) {
-                        logger.log(Messages.Builder_Polling_Timeout_Exceeded());
-                        return;
-                    }
-                    logger.log(Messages.Builder_Polling());
-                }
-                if (this.projectId == null) {
-                    // project was auto-created. Fetch it's new uuid so that we can look up the results
-                    logger.log(Messages.Builder_Project_Lookup());
-                    final String jsonResponseBody = apiClient.lookupProject(this.projectName, this.projectVersion);
-                    this.projectId = new ProjectLookupParser(jsonResponseBody).parse().getProject().getUuid();
+                if (timeout < System.currentTimeMillis()) {
+                    logger.log(Messages.Builder_Polling_Timeout_Exceeded());
+                    // XXX this seems like a fatal error
+                    throw new AbortException("Dependency Track server response timeout");
```

Comment:
It seemed strange that this wasn't being treated as a build failure.  If we don't hear back from the deptrack server about whether this build has CVEs or not, seems like we should halt the build.